File: client/src/app/+models/learning_resource_model.ts
import { LearningResource } from './dtos/learning_resource_dto';
import { Post } from './dtos/post_dto';
import { UserResourceState } from './dtos/user_resource_state_dto';

export interface LearningResourceModel extends LearningResource {
    posts: Post[];
    userResourceState: UserResourceState;
}

File: server/BitBoard/2_Business/BitBoard.Business.Views/Learning/ViewModels/LearningResourceModel.cs
using System.Collections.Generic;
using BitBoard.Business.Views.Learning.Dtos;

namespace BitBoard.Business.Views.Learning.ViewModels
{
    public class LearningResourceModel : LearningResourceDto
    {
        public ICollection<PostDto> Posts { get; set; }
        public UserResourceStateDto UserResourceProgression { get; set; }
    }
}